dashboard/src/components/FunctionView/FunctionView.tsx

```diff
@@ -74,7 +74,7 @@ class FunctionView extends React.Component<IFunctionViewProps, IFunctionViewStat
     // refetch pod name when deployment is available, in case in changed
     // TODO: move deployment status into redux store
     const status: IDeploymentStatus = deployment.status;
-    if (status.availableReplicas !== status.replicas) {
+    if (!status.availableReplicas || status.availableReplicas !== status.replicas) {
```

Comment:
there was a scenario where `status.availableReplicas === status.replicas` because both were `undefined`, and it was getting marked as healthy! This stops that.